virridy/non_targets_workflow/00_virridy_preparatory_workflow.Rmd

```diff
@@ -0,0 +1,254 @@
+time*
+
+```{r}
+old_field_notes <- load_old_field_notes(filepath = "data/sensor_field_notes.xlsx")
+
+new_field_notes <- load_mWater_notes() %>%
+  grab_mWater_sensor_notes(mWater_api_data = .)
+
+#merge new mwater notes (sensor_notes) and old notes (field notes)  
+all_field_notes <- rbind(old_field_notes, new_field_notes) %>%
+  mutate(site = ifelse(site == "riverbluffs", "river bluffs", site))
+
+sensor_malfunction_notes <- grab_mWater_malfunction_notes(mWater_api_data = load_mWater_notes()) 
+```
+
+*Merge the data sets from all API pulls. Then for developing this workflow, subset the
+data to only the 2022 and 2023 field season. (Previous field seasons were managed quite
+differently, and therefore should be treated differently.)*
+
+```{r}
+hv_creds <- read_yaml("creds/HydroVuCreds.yml")
+
+hv_token <- hv_auth(client_id = as.character(hv_creds["client"]),
+                    client_secret = as.character(hv_creds["secret"]),
+                    url = "https://www.hydrovu.com/public-api/oauth/token")
+
+# all_older_data <- try(munge_api_data(api_path = "data/api/virridy/", network = "virridy") %>%
+#   dplyr::filter(lubridate::year(DT_round) >= 2023))
+
+incoming_data_csvs_upload <- api_puller(site = "Timberline", #c("JOEI", "CBRI", "CHD", "PFAL", "PBD", "SFM",
+                                                 #"LBEA", "PENN", "BoxCreek", "Timberline", "Prospect", "SpringCreek", "Archery"),
+                                        start_dt = "2023-01-01 01:00:00 MDT",
+                                        end_dt = Sys.time(), 
+                                        api_token = hv_token,
+                                        dump_dir = "data/api/virridy/",
+                                        require = NULL)
+
+incoming_data_collated_csvs <- munge_api_data(api_path = "data/api/virridy/",
+                                              network = "virridy",
+                                              require = NULL)
+all_data <- incoming_data_collated_csvs
+```
+
+## *Step 2: Develop site-parameter data thresholds*
+
+*Here, we split up all of our site-parameter combinations into a list that we can more
+easily iterate over. Then, across those lists, we average any observations whose frequency
+is greater than 15 minutes so that our data set is consistently recorded at 15-minute
+intervals. We also preserve the total number of observations within the 15-minute
+increment used to calculate the mean, as well as the spread (max-min). After these
+calculations, we use {padr}'s `pad()` function to fill in data gaps at this 15-minute
+interval. Lastly, we join these data frames with the field notes.*
+
+```{r}
+# format and summarize data
+# Determine each site and parameter in all_data 
+sites <- unique(all_data$site)
+params <- c(# "Battery Level",
+  # "Baro",
+  "Chl-a Fluorescence", 
+  "Depth", 
+  "DO", 
+  # "External Voltage", 
+  "ORP", 
+  "pH",
+  "Specific Conductivity",
+  "Temperature",
+  "Turbidity",
+  "FDOM")
```

Comment:
It doesn't appear that any of the FDOM data got saved. I think this needs to be "FDOM Fluorescence"

Reply:
You're right! Re-running now... should be updated in the data folder quickly, since it was still downloaded in the API pull. :-)